### PR TITLE
Refactor the logic, make it more generic

### PR DESCRIPTION
**Pull request by DarkoKukovec**
* Detect the project type and info
* Handle next.js special cases
* Document all the "automagic" logic
* Fix Slack notifications
* Document all TODOs

### CODE REVIEW

**Comment by DarkoKukovec** [comment]
Once this is confirmed, I'll open tasks in Productive for all mentioned TODOs

**Review by kristian240 — APPROVED** [review]

File: .github/actions/bootstrap/action.yml [hunk 1]
```diff
@@ -56,3 +61,14 @@ runs:
         # If source files changed but packages didn't, rebuild from a prior cache.
         restore-keys: |
           ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') || hashFiles('**/yarn.lock') }}-
+    - name: Cache node modules for others
+      uses: actions/cache@v3
+      if: ${{ inputs.framework != 'next' }}
+      with:
+        path: |
```
Comment (mislavk) [inline .github/actions/bootstrap/action.yml:68]: Is this necessary for oneliner? Is this preparation for another command?

File: .github/actions/bootstrap/action.yml [hunk 2]
```diff
@@ -56,3 +61,14 @@ runs:
...
+          node_modules
+        # Generate a new cache whenever packages or source files change.
+        key: ${{ runner.os }}-app-${{ hashFiles('**/package-lock.json') || hashFiles('**/yarn.lock') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
+        # If source files changed but packages didn't, rebuild from a prior cache.
+        restore-keys: |
+          ${{ runner.os }}-app-${{ hashFiles('**/package-lock.json') || hashFiles('**/yarn.lock') }}-
```
Comment (mislavk) [inline .github/actions/bootstrap/action.yml:74]:
```suggestion
          ${{ runner.os }}-app-${{ hashFiles('**/package-lock.json') || hashFiles('**/yarn.lock') }}
```

**Review by mislavk — APPROVED** [review]

File: .github/actions/bootstrap/action.yml (hunk 1 above)
Comment (DarkoKukovec) [reply .github/actions/bootstrap/action.yml]: Probably not 😅

**Pull request merged**